pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -335,7 +338,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env *model.En
 			}
 
 			pluginParams := &plugin.InputParams{
-				ListenerProtocol:           plugin.ModelProtocolToListenerProtocol(endpoint.ServicePort.Protocol),
+				ListenerProtocol:           plugin.ModelProtocolToListenerProtocol(node, endpoint.ServicePort.Protocol),
```

Comment:
This (and below) would return `ListenerProtocolAuto` for unsupported protocol, please be reminded this will break the authentication and authorization plugin, see the code reference in my other comment. Could you add a TODO about fixing this for authN/authZ before releasing this feature? Thank you.

Reply:
TODO added.